AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -131,26 +144,132 @@ + (void)handleUrlResponse:(NSURL *)url {
-    return;
++ (void)login {
+
+  // TODO protect with canBeUsed.
+  [[MSIdentity sharedInstance] login];
+}
+
+- (void)login {
+  @synchronized(lock) {
+    if (self.clientApplication == nil && self.identityConfig == nil) {
+      self.loginDelayed = YES;
+      return;
+    }
+    self.loginDelayed = NO;
+    [self.clientApplication acquireTokenForScopes:@[ (NSString * _Nonnull) self.identityConfig.identityScope ]
+                                  completionBlock:^(MSALResult *result, NSError *e) {
+                                    // TODO: Implement error handling.
+                                    if (e) {
+                                    } else {
+                                      NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
+                                      self.accessToken = result.accessToken;
+                                    }
+                                  }];
   }
-  NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:kMSAuthorityFormat, kMSTenantName, kMSPolicyName]];
-  MSALAuthority *authority = [MSALAuthority authorityWithURL:url error:nil];
-  [self.clientApplication acquireTokenForScopes:@[ [NSString stringWithFormat:kMSAuthScopeFormat, kMSTenantName, kMSAPIIdentifier] ]
-                           extraScopesToConsent:nil
-                                      loginHint:nil
-                                     uiBehavior:MSALUIBehaviorDefault
-                           extraQueryParameters:nil
-                                      authority:authority
-                                  correlationId:nil
-                                completionBlock:^(MSALResult *result, NSError *e) {
-                                  // TODO: Implement error handling.
-                                  if (e) {
-                                  } else {
-                                    NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
-                                    self.accessToken = result.accessToken;
-                                  }
-                                }];
 }
+
+#pragma mark - Private methods
+
+- (NSString *)identityConfigFilePath {
+  return [NSString stringWithFormat:@"%@/%@", kMSIdentityPathComponent, kMSIdentityConfigFilename];
+}
+
+- (BOOL)loadConfigurationFromCache {
+  NSData *configData = [MSUtility loadDataForPathComponent:[self identityConfigFilePath]];
+  self.identityConfig = [self deserializeData:configData];
+  if (self.identityConfig == nil || ![self.identityConfig isValid]) {
+    [self clearConfigurationCache];
+    self.identityConfig = nil;
+    return NO;
+  }
+  return YES;
+}
+
+- (void)downloadConfigurationWithETag:(nullable NSString *)eTag {
+
+  // Download configuration.
+  MSIdentityConfigIngestion *ingestion =
+      [[MSIdentityConfigIngestion alloc] initWithBaseUrl:@"https://mobilecentersdkdev.blob.core.windows.net" appSecret:self.appSecret];
```

Comment:
I think this URL string should be in a constant

Reply:
This is a temporary. We don't have an endpoint right now so it will be removed once we have it.

Reply:
So, will keep it.

Reply:
No problem